botocore/credentials.py

```diff
@@ -1295,30 +1367,48 @@ def _get_role_config(self, profile_name):
         else:
             self._validate_source_profile(profile_name, source_profile)
 
-        return role_config
-
     def _validate_credential_source(self, parent_profile, credential_source):
         if self._credential_sourcer is None:
-            raise InvalidConfigError(error_msg=(
-                'The credential_source "%s" is specified in profile "%s", '
-                'but no source provider was configured.' % (
-                    credential_source, parent_profile)
-            ))
-        if not self._credential_sourcer.is_supported(credential_source):
-            raise InvalidConfigError(error_msg=(
-                'The credential source "%s" referenced in profile "%s" is not '
-                'valid.' % (credential_source, parent_profile)
-            ))
+            if parent_profile is not None:
+                error_msg = (
```

Comment:
This is a lot of code and branches just to get nice error messages, is there any other way you might have me write this?